plugins/node/opentelemetry-instrumentation-undici/test/utils/assertSpan.ts

```diff
@@ -0,0 +1,199 @@
+      span.attributes[SemanticAttributes.HTTP_RESPONSE_STATUS_CODE]
+    }`
+  );
+
+  assert.strictEqual(span.links.length, 0, 'there are no links');
+
+  if (validations.error) {
+    assert.strictEqual(span.events.length, 1, 'span contains one error event');
+    assert.strictEqual(
+      span.events[0].name,
+      'exception',
+      'error event name is correct'
+    );
+
+    const eventAttributes = span.events[0].attributes;
+    assert.ok(eventAttributes != null, 'event has attributes');
+    assert.deepStrictEqual(
+      Object.keys(eventAttributes),
+      ['exception.type', 'exception.message', 'exception.stacktrace'],
+      'the event attribute names are correct'
+    );
+  } else {
+    assert.strictEqual(span.events.length, 0, 'span contains no events');
+  }
+
+  // Error message changes between version se we will
+  // only assert its presence
+  if (validations.forceStatus) {
+    assert.equal(
+      span.status.code,
+      validations.forceStatus.code,
+      'span `status.code` is correct'
+    );
+    assert.ok(span.status.message, 'span `status.message` is present');
+  } else {
+    const { httpStatusCode } = validations;
+    const isStatusUnset =
+      httpStatusCode && httpStatusCode >= 100 && httpStatusCode < 400;
+    assert.equal(
+      span.status.code,
+      isStatusUnset ? SpanStatusCode.UNSET : SpanStatusCode.ERROR,
+      'span `status.code` is correct'
+    );
+  }
+
+  assert.ok(span.endTime, 'must be finished');
+  assert.ok(
+    hrTimeToNanoseconds(span.duration) > 0,
+    'must have positive duration'
+  );
+
+  if (validations.resHeaders) {
+    // Headers were added in v17.5.0, v16.15.0
+    // https://nodejs.org/api/globals.html#class-headers
+    const { resHeaders } = validations;
+    const contentLengthHeader = getHeader(resHeaders, 'content-length');
+
+    if (contentLengthHeader) {
+      const contentLength = Number(contentLengthHeader);
+
+      assert.strictEqual(
+        span.attributes['http.response.header.content-length'],
+        contentLength
+      );
+    }
+  }
+
+  assert.strictEqual(
+    span.attributes[SemanticAttributes.SERVER_ADDRESS],
+    validations.hostname,
+    'must be consistent (SERVER_ADDRESS and hostname)'
+  );
+  if (!validations.noNetPeer) {
+    assert.ok(
+      span.attributes[SemanticAttributes.NETWORK_PEER_ADDRESS],
+      `must have ${SemanticAttributes.NETWORK_PEER_ADDRESS}`
+    );
+    assert.ok(
+      span.attributes[SemanticAttributes.NETWORK_PEER_PORT],
+      `must have ${SemanticAttributes.NETWORK_PEER_PORT}`
+    );
+  }
+  assert.ok(
+    (span.attributes[SemanticAttributes.URL_FULL] as string).indexOf(
+      span.attributes[SemanticAttributes.SERVER_ADDRESS] as string
+    ) > -1,
+    `${SemanticAttributes.URL_FULL} & ${SemanticAttributes.SERVER_ADDRESS} must be consistent`
+  );
+
+  if (validations.reqHeaders) {
+    const userAgent = getHeader(validations.reqHeaders, 'content-length');
```

Comment:
Hmm, is that `content-length` being checked against user-agent below? :thinking:

Reply:
this and commented lines below is a test update I did not finished. I?ll have it updated in next commit

Reply:
done